test/parallel_api/algorithm/alg.merge/inplace_merge.pass.cpp

```diff
@@ -56,29 +80,47 @@ struct test_one_policy
 
 template <typename T, typename Generator1, typename Generator2, typename Compare>
 void
-test_by_type(Generator1 generator1, Generator2 generator2, Compare comp)
+test_by_type(Generator1 generator1, Generator2 generator2, bool comp_flag, Compare comp)
 {
     using namespace std;
     size_t max_size = 100000;
     Sequence<T> in1(max_size, [](size_t v) { return T(v); });
     Sequence<T> exp(max_size, [](size_t v) { return T(v); });
     size_t m;
 
-    for (size_t n = 0; n <= max_size; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+    if(comp_flag)
     {
-        m = 0;
-        invoke_on_all_policies<0>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
-                               generator1, generator2, comp);
-
-        m = n / 3;
-        invoke_on_all_policies<1>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
-                               generator1, generator2, comp);
-
+        for (size_t n = 0; n <= max_size; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+        {
+            m = 0;
+            invoke_on_all_policies<0>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
+                                   generator1, generator2, comp);
+            m = n / 3;
+            invoke_on_all_policies<1>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
+                                   generator1, generator2, comp);
+#if !ONEDPL_FPGA_DEVICE
+            m = 2 * n / 3;
+            invoke_on_all_policies<2>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
+                                   generator1, generator2, comp);
+#endif
+        }
+    }
+    else
+    {
+        for (size_t n = 0; n <= max_size; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+        {
+            m = 0;
+            invoke_on_all_policies<3>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
+                                   generator1, generator2);
+            m = n / 3;
+            invoke_on_all_policies<4>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
+                                   generator1, generator2);
 #if !ONEDPL_FPGA_DEVICE
-        m = 2 * n / 3;
-        invoke_on_all_policies<2>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
-                               generator1, generator2, comp);
+            m = 2 * n / 3;
+            invoke_on_all_policies<5>()(test_one_policy<T>(), in1.begin(), in1.begin() + n, exp.begin(), exp.begin() + n, n, m,
+                                   generator1, generator2);
 #endif
+        }
```

Comment:
Now it's better than previous version, but I still see a duplication code here. I think it would be better if `if` was inside the `for` loop over each `invoke_on_all_policies`.

Reply:
Improved